eng/common/TestResources/New-TestResources.ps1

```diff
@@ -101,6 +108,16 @@ function Retry([scriptblock] $Action, [int] $Attempts = 5) {
     }
 }
 
+function MergeHashes([hashtable] $source, [psvariable] $dest) {
+    foreach ($key in $source.Keys) {
+        if ($dest.Value[$key]) {
```

Comment:
Instead of passing in the variable name why not pass in the dest as a hashtable as well? I would also suggest you use ".ContainsKey" here instead of indexing into the value directly.

Reply:
I wanted to be able to log the name of the hash so a log reader could understand which object had overrides against it (e.g. deployment outputs, template parameters, etc.). I think the two options for that are to pass a third variable representing the name, or do it this way.